services/cache/lru_striped.go

```diff
@@ -0,0 +1,153 @@
+// each LRU bucket is purged after another one, which means that keys could still
+// be present after a call to Purge.
+func (L LRUStriped) Purge() error {
+	for _, lru := range L.buckets {
+		lru.Purge() // errors from purging LRU can be ignored as they always return nil
+	}
+	return nil
+}
+
+// Set does the same as LRU.Set
+func (L LRUStriped) Set(key string, value interface{}) error {
+	return L.keyBucket(key).Set(key, value)
+}
+
+// SetWithDefaultExpiry does the same as LRU.SetWithDefaultExpiry
+func (L LRUStriped) SetWithDefaultExpiry(key string, value interface{}) error {
+	return L.keyBucket(key).SetWithDefaultExpiry(key, value)
+}
+
+// SetWithExpiry does the same as LRU.SetWithExpiry
+func (L LRUStriped) SetWithExpiry(key string, value interface{}, ttl time.Duration) error {
+	return L.keyBucket(key).SetWithExpiry(key, value, ttl)
+}
+
+// Get does the same as LRU.Get
+func (L LRUStriped) Get(key string, value interface{}) error {
+	return L.keyBucket(key).Get(key, value)
+}
+
+// Remove does the same as LRU.Remove
+func (L LRUStriped) Remove(key string) error {
+	return L.keyBucket(key).Remove(key)
+}
+
+// Keys does the same as LRU.Keys. However, because this is lock-free, keys might be
+// inserted or removed from a previously scanned LRU cache.
+// This is not as precise as using a single LRU instance.
+func (L LRUStriped) Keys() ([]string, error) {
+	keys := make([]string, 0)
+	for _, lru := range L.buckets {
+		k, _ := lru.Keys() // Keys never returns any error
+		keys = append(keys, k...)
+	}
+	return keys, nil
+}
+
+// Len does the same as LRU.Len. As for LRUStriped.Keys, this call cannot be precise.
+func (L LRUStriped) Len() (int, error) {
+	size := 0
+	for _, lru := range L.buckets {
+		s, _ := lru.Len() // Len never returns any error
+		size += s
+	}
+	return size, nil
+}
+
+// GetInvalidateClusterEvent does the same as LRU.GetInvalidateClusterEvent
+func (L LRUStriped) GetInvalidateClusterEvent() string {
+	return L.buckets[0].GetInvalidateClusterEvent()
+}
+
+// Name does the same as LRU.Name
+func (L LRUStriped) Name() string {
+	return L.buckets[0].Name()
+}
+
+// NewLRUStriped creates a striped LRU cache using the special LRUOptions.StripedBuckets value.
+// See LRUStriped and LRUOptions for more details.
+//
+// Not that in order to prevent false eviction, this LRU cache adds 10% (computation is rounded up) of the
+// requested size to the total cache size.
+func NewLRUStriped(opts LRUOptions) Cache {
+	if opts.StripedBuckets == 0 {
+		opts.StripedBuckets = runtime.NumCPU()
+	}
+
+	if opts.Size < opts.StripedBuckets {
+		opts.StripedBuckets = opts.Size
+	}
+
+	if opts.Size == 0 || opts.StripedBuckets == 0 {
+		opts.StripedBuckets = 1
+	}
+
+	// add 10% to the total size, before splitting
+	opts.Size += int(math.Ceil(float64(opts.Size) * 10.0 / 100.0))
+	// now this is the size for each bucket
+	opts.Size = (opts.Size / opts.StripedBuckets) + (opts.Size % opts.StripedBuckets)
+
+	buckets := make([]*LRU, 0)
```

Comment:
specify capacity of `len(opts.StripedBuckets)`

Reply:
done